python/cudf/cudf/core/column/numerical.py

```diff
@@ -165,24 +169,178 @@ def as_numerical_column(self, dtype, **kwargs):
             return self
         return libcudf.unary.cast(self, dtype)
 
-    def sum(self, dtype=None):
-        return libcudf.reduce.reduce("sum", self, dtype=dtype)
+    def sum(self, skipna=None, dtype=None, min_count=0):
+        result_col = self._process_for_reduction(
+            skipna=skipna, min_count=min_count
+        )
+        if isinstance(result_col, cudf.core.column.ColumnBase):
+            return libcudf.reduce.reduce("sum", result_col, dtype=dtype)
+        else:
+            return result_col
 
-    def product(self, dtype=None):
-        return libcudf.reduce.reduce("product", self, dtype=dtype)
+    def product(self, skipna=None, dtype=None, min_count=0):
+        result_col = self._process_for_reduction(
+            skipna=skipna, min_count=min_count
+        )
+        if isinstance(result_col, cudf.core.column.ColumnBase):
+            return libcudf.reduce.reduce("product", result_col, dtype=dtype)
+        else:
+            return result_col
 
-    def mean(self, dtype=np.float64):
-        return libcudf.reduce.reduce("mean", self, dtype=dtype)
+    def mean(self, skipna=None, dtype=np.float64):
+        result_col = self._process_for_reduction(skipna=skipna)
+        if isinstance(result_col, cudf.core.column.ColumnBase):
+            return libcudf.reduce.reduce("mean", result_col, dtype=dtype)
+        else:
+            return result_col
 
-    def var(self, ddof=1, dtype=np.float64):
-        return libcudf.reduce.reduce("var", self, dtype=dtype, ddof=ddof)
+    def var(self, skipna=None, ddof=1, dtype=np.float64):
+        result = self._process_for_reduction(skipna=skipna)
+        if isinstance(result, cudf.core.column.ColumnBase):
+            return libcudf.reduce.reduce("var", result, dtype=dtype, ddof=ddof)
+        else:
+            return result
 
-    def std(self, ddof=1, dtype=np.float64):
-        return libcudf.reduce.reduce("std", self, dtype=dtype, ddof=ddof)
+    def std(self, skipna=None, ddof=1, dtype=np.float64):
+        result_col = self._process_for_reduction(skipna=skipna)
+        if isinstance(result_col, cudf.core.column.ColumnBase):
+            return libcudf.reduce.reduce(
+                "std", result_col, dtype=dtype, ddof=ddof
+            )
+        else:
+            return result_col
 
     def sum_of_squares(self, dtype=None):
         return libcudf.reduce.reduce("sum_of_squares", self, dtype=dtype)
 
+    def kurtosis(self, skipna=None):
+        skipna = True if skipna is None else skipna
+
+        if len(self) == 0 or (not skipna and self.has_nulls):
+            return cudf.utils.dtypes._get_nan_for_dtype(self.dtype)
+
+        self = self.nans_to_nulls().dropna()
+
+        if len(self) < 4:
+            return cudf.utils.dtypes._get_nan_for_dtype(self.dtype)
+
+        n = len(self)
+        miu = self.mean()
+        m4_numerator = (
+            self.binary_operator("sub", miu).binary_operator(
```

Comment:
Just a note we should really implement things like `__sub__` at the column level to avoid having to use the `binary_operator` API everywhere.

Reply:
Sure, will track it with another issue

Reply:
Added an issue for it: https://github.com/rapidsai/cudf/issues/6192